barretenberg/cpp/src/barretenberg/crypto/poseidon2/poseidon2.cpp

```diff
@@ -5,10 +5,9 @@ namespace crypto {
  * @brief Hashes a vector of field elements
  */
 template <typename Params>
-typename Poseidon2<Params>::FF Poseidon2<Params>::hash(const std::vector<typename Poseidon2<Params>::FF>& input)
+typename Poseidon2<Params>::FF Poseidon2<Params>::hash(const std::span<typename Poseidon2<Params>::FF>& input)
 {
-    auto input_span = input;
```

Comment:
I think this input_span is still a vector, but not completely sure how this inference works.

Reply:
it's just the type of the right hand side, typeof(input) is std::vector so auto becomes std::vector and like a typical std::vector a = b it copies

Reply:
ah, ok so the function call on the next line doesn't have an effect on this inference.